packages/flutter_tools/lib/src/debug_adapters/flutter_adapter.dart

```diff
@@ -397,8 +397,25 @@ class FlutterDebugAdapter extends DartDebugAdapter<FlutterLaunchRequestArguments
   /// Connects to the VM Service if the app.started event has fired, and a VM Service URI is available.
   void _connectDebuggerIfReady() {
     final Uri? serviceUri = _vmServiceUri;
+
     if (_receivedAppStarted && serviceUri != null) {
-      connectDebugger(serviceUri, resumeIfStarting: true);
+      if (enableDebugger) {
+        connectDebugger(serviceUri, resumeIfStarting: true);
+      } else {
+        // Usually, `connectDebugger` (in the base Dart adapter) will send this
+        // event when it connects a debugger. Since we're not connecting a
+        // debugger we send this ourselves, to allow clients to connect to the
+        // VM Service for things like starting DevTools, even if debugging is
+        // not available.
+        // TODO(dantup): Switch this to call `sendDebuggerUris()` on the base
+        //   adapter once rolled into Flutter.
+        sendEvent(
+          RawEventBody(<String, Object?>{
+            'vmServiceUri': serviceUri.toString(),
+          }),
+          eventType: 'dart.debuggerUris',
+        );
```

Comment:
I have a CL that adds a method to the base to avoid duplicating it here, but it needs a new DDS release (and then rolling into here). Assuming that doesn't happen before this is ready, I'll sort this in a separate PR once it does.

Reply:
are you saying the DDS change hasn't been published yet?

Reply:
It's in DDS 2.2.3 published today, however that can't be rolled into Flutter yet due to a conflicting vm_service constraint with DWDS.

In case it isn't clear, the DDS change is _not_ required for this code to function, only to replace lines 412-417 with a simpler base method call so that the Flutter adapter doesn't need to construct the `dart.debuggerUris` message itself.

Reply:
> I'm guessing it's (going to be) [this](https://github.com/dart-lang/sdk/commit/c75f794913a398248b15f5b0e2933f93917b43c3)

Yup! The real change was this, which is just extracting a method:

https://github.com/dart-lang/sdk/commit/fe8dd70d4751a0184f9b399a1ec188407c775d65